lld/wasm/SyntheticSections.cpp

```diff
@@ -568,6 +568,13 @@ void ElemSection::addEntry(FunctionSymbol *sym) {
     return;
   sym->setTableIndex(ctx.arg.tableBase + indirectFunctions.size());
   indirectFunctions.emplace_back(sym);
```

Comment:
The problem here is that the very first function will not be aligned here since `tableBase` defaults to `1` which is not aligned.

How about doing this before the call to `indirectFunctions.emplace_back(sym);` (i.e. do the alignment before placing the function in the table instead of after?  That should also avoid the trailing null function entries in the table, that serve no purpose.

Reply:
Thanks for pointing this out - I've gotten this fixed and the test corrected